libstuff/SHTTPSManager.cpp

```diff
@@ -222,8 +173,10 @@ SStandaloneHTTPSManager::Transaction* SStandaloneHTTPSManager::_httpsSend(const
 
     // If this is going to be an https transaction, create a certificate and give it to the socket.
     SX509* x509 = SStartsWith(url, "https://") ? SX509Open(_pem, _srvCrt, _caCrt) : nullptr;
-    Socket* s = openSocket(host, x509);
-    if (!s) {
+    Socket* s = nullptr;
+    try {
```

Comment:
Because we use a Socket constructor instead of a method that returns a pointer to a new socket, we have to handle a socket failing to open with an exception.